### PR TITLE
Add API based modules by default

### PR DESCRIPTION
**Pull request by databyjp**
### What's being changed:

Enable API-based t2v, generative & reranker modules by default

Please free to refactor as you see fit :) 

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: adapters/handlers/rest/configure_api.go [hunk 1]
```diff
@@ -696,12 +696,47 @@ func registerModules(appState *state.State) error {
...
+		"generative-cohere",
+		"generative-mistral",
+		"generative-octoai",
+		"generative-openai",
+		"generative-palm",
+	}
+	defaultOthers := []string{
+		"reranker-cohere",
+		"reranker-voyageai",
+	}
```
Comment (antas-marcin) [inline adapters/handlers/rest/configure_api.go:722]: Please use `.Name` constants to get the name of the module (eg. `modgenerativemistral.Name`) to mitigate the misspellings.

File: adapters/handlers/rest/configure_api.go [hunk 2]
```diff
@@ -696,12 +696,47 @@ func registerModules(appState *state.State) error {
...
+
 	if len(appState.ServerConfig.Config.EnableModules) > 0 {
-		modules := strings.Split(appState.ServerConfig.Config.EnableModules, ",")
-		for _, module := range modules {
-			enabledModules[strings.TrimSpace(module)] = true
-		}
+		modules = strings.Split(appState.ServerConfig.Config.EnableModules, ",")
+	} else {
+		modules = []string{}
+	}
```
Comment (antas-marcin) [inline adapters/handlers/rest/configure_api.go:731]: the `modules` variable doesn't need to be initialized bc `append(modules, ...)` is able to handle nil slices

File: adapters/handlers/rest/configure_api.go [hunk 3]
```diff
@@ -696,12 +696,47 @@ func registerModules(appState *state.State) error {
...
+		"generative-octoai",
+		"generative-openai",
+		"generative-palm",
+	}
+	defaultOthers := []string{
+		"reranker-cohere",
+		"reranker-voyageai",
+	}
+
 	enabledModules := map[string]bool{}
```
Comment (antas-marcin) [inline adapters/handlers/rest/configure_api.go:724]: I would the declaration of `enabledModules` map to line just above 738

File: adapters/handlers/rest/configure_api.go (hunk 1 above)
Comment (databyjp) [reply adapters/handlers/rest/configure_api.go]: Updated

File: adapters/handlers/rest/configure_api.go (hunk 2 above)
Comment (databyjp) [reply adapters/handlers/rest/configure_api.go]: Ty! Updated

File: adapters/handlers/rest/configure_api.go (hunk 3 above)
Comment (databyjp) [reply adapters/handlers/rest/configure_api.go]: Makes sense. Updated.

File: adapters/handlers/rest/configure_api.go [hunk 4]
```diff
@@ -696,12 +697,46 @@ func registerModules(appState *state.State) error {
...
+		modgenerativeoctoai.Name,
+		modgenerativeopenai.Name,
+		modgenerativepalm.Name,
+	}
+	defaultOthers := []string{
+		modrerankercohere.Name,
+		modrerankervoyageai.Name,
+	}
+
+	defaultModules := append(defaultVectorizers, defaultGenerative...)
```
Comment (antas-marcin) [inline adapters/handlers/rest/configure_api.go:725]: I think you forgot about the defaultVectorizers...

**Review by antas-marcin — APPROVED** [review]

**Pull request merged**